dataset/dataset.py

```diff
@@ -1,31 +1,37 @@
 import copy
 import gzip
+import json
 import torch
-from torch.utils.data import Dataset, DataLoader, IterableDataset
-from transformers import BertTokenizer
+import math
+from torch.utils.data import Dataset, IterableDataset
+from numpy.random import choice
 
 """
 Implementation of lazy dataloder in PyTorch
 """
 
 class TextIterator:
-  def __init__(self, text_iterator, batch_size, num_workers, transform=None):
+  def __init__(self, text_iterator, batch_size, lengths, weights, num_workers,
+               transform=None):
     self.batch_size = batch_size
     self.iter_number = 0
     self.num_workers = num_workers
     self.text_iterator = text_iterator
     self.transform = transform
+    self.lengths = lengths
+    self.weights = weights
 
   def __iter__(self):
     return self.text_iterator
 
   def __next__(self):
+    sampled_dataset = choice(len(self.lengths), p=self.weights)
     if self.iter_number == self.batch_size:
```

Comment:
Here you should check whether you need to skip lines given the number of iterations performed on the given dataset so something like if self.iter_number[sampled_dataset] == self.batch_size: